### ISSUE TITLE
ProcessOutput.Readlines gets stuck forever

### ISSUE THREAD

**Issue by DavidPerezIngeniero** [com-lihaoyi/os-lib#199]
Simple snippet using Amminite REPL:
```
Using (Files.newBufferedWriter((os.home / s"kk.txt").toNIO)) { w =>
    os.proc("ps", "-ef").call(stdout=os.ProcessOutput.Readlines(l => w.write(l)))
  } 
```

Call stack:
```
"main" #1 prio=5 os_prio=0 cpu=3484,79ms elapsed=898,29s tid=0x00007fd144026a80 nid=0x7b28 in Object.wait()  [0x00007fd14db85000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(java.base@17.0.7/Native Method)
        - waiting on <0x00000000e3c01020> (a java.lang.Thread)
        at java.lang.Thread.join(java.base@17.0.7/Thread.java:1304)
        - locked <0x00000000e3c01020> (a java.lang.Thread)
        at java.lang.Thread.join(java.base@17.0.7/Thread.java:1372)
        at os.SubProcess.$anonfun$join$1(SubProcess.scala:73)
        at os.SubProcess.$anonfun$join$1$adapted(SubProcess.scala:73)
        at os.SubProcess$$Lambda$1674/0x00000008015e6b50.apply(Unknown Source)
        at scala.Option.foreach(Option.scala:437)
        at os.SubProcess.join(SubProcess.scala:73)
        at os.proc.call(ProcessOps.scala:80)
        at ammonite.$sess.cmd2$.$anonfun$res2$2(cmd2.sc:2)
        at ammonite.$sess.cmd2$$$Lambda$1658/0x0000000801598ce0.apply(Unknown Source)
        at scala.util.Using$.$anonfun$apply$1(Using.scala:261)
        at scala.util.Using$.apply(Using.scala:113)
        at ammonite.$sess.cmd2$.<clinit>(cmd2.sc:1)
        at ammonite.$sess.cmd2.$main(cmd2.sc)
```

Environment:
- Ammonite Repl 2.5.2 
- Scala 2.13.8 
- Java 17.0.7
- Linux

**Comment by DavidPerezIngeniero** [comment]
It looks like a duplicate of https://github.com/com-lihaoyi/os-lib/issues/30
